frontend/services/entities/mdm_android.ts

```diff
@@ -24,4 +25,46 @@ export default {
     const { MDM_ANDROID_ENTERPRISE } = endpoints;
     return sendRequest("DELETE", MDM_ANDROID_ENTERPRISE);
   },
+
+  /**
+   * This function starts a Server-Sent Events connection with the fleet server
+   * to get messages about a successful Android mdm connection. We have to use
+   * fetch here because the EventSource API does not support setting headers,
+   * which we need to authenticate the request.
+   */
+  startSSE: (abortSignal: AbortSignal): Promise<void> => {
+    return new Promise(async (resolve, reject) => {
+      try {
+        const response = await fetch(endpoints.MDM_ANDROID_SSE_URL, {
+          method: "GET",
+          headers: {
+            Authorization: `Bearer ${authToken()}`,
+          },
+          signal: abortSignal,
+        });
+
+        const reader = response?.body?.getReader();
+        const decoder = new TextDecoder();
+
+        while (true) {
+          // @ts-ignore
+          // eslint-disable-next-line no-await-in-loop
+          const { done, value } = await reader?.read();
```

Comment:
Just curious, how does `await` handle the case were `reader` is undefined?

Reply:
it will fail and the catch block will run. if reader is undefined there was an issue with the request up above. ultimately we will show the error flash message if this is the case.